freqtrade/edge/__init__.py

```diff
@@ -0,0 +1,388 @@
+# pragma pylint: disable=W0603
+""" Edge positioning package """
+import logging
+from typing import Any, Dict
+import arrow
+
+import numpy as np
+import utils_find_1st as utf1st
+from pandas import DataFrame
+
+import freqtrade.optimize as optimize
+from freqtrade.optimize.backtesting import BacktestResult
+from freqtrade.arguments import Arguments
+from freqtrade.exchange import Exchange
+from freqtrade.strategy.interface import SellType
+from freqtrade.strategy.resolver import IStrategy, StrategyResolver
+from freqtrade.optimize.backtesting import Backtesting
+
+
+logger = logging.getLogger(__name__)
+
+
+class Edge():
+
+    config: Dict = {}
+
+    def __init__(self, config: Dict[str, Any], exchange=None) -> None:
+        """
+        constructor
+        """
+        self.config = config
+        self.strategy: IStrategy = StrategyResolver(self.config).strategy
+        self.ticker_interval = self.strategy.ticker_interval
+        self.tickerdata_to_dataframe = self.strategy.tickerdata_to_dataframe
+        self.get_timeframe = Backtesting.get_timeframe
+        self.advise_sell = self.strategy.advise_sell
+        self.advise_buy = self.strategy.advise_buy
+
+        self.edge_config = self.config.get('edge', {})
+
+        self._last_updated = None
+        self._cached_pairs: list = []
+        self._total_capital = self.edge_config['total_capital_in_stake_currency']
+        self._allowed_risk = self.edge_config['allowed_risk']
+
+        ###
+        #
+        ###
+        if exchange is None:
```

Comment:
this check (and defaulting the exchange to `None` in the function header) does not make sense - `Edge `is only instanciated once as far as i can see - and that's 1 line after instanciating `exchange` and then passing it to the constructor.